### ISSUE TITLE
Timestamp for each capability/sensor in a device

### ISSUE THREAD

**Issue by ewikner** [athombv/homey#974]
**What is your Homey version (Settings → System)?**
0.10.3

**What did you try to achieve?**
Add a timestamp for each sensor/capability in a device that will show the time when it got a value like "lastupdate". I will use it so know if a sensor is dead.
To make this more dynamic so other developers can use it for there apps the best will be if you can add it to the device object and then be able to update it for each capability/sensor.

Here is an other input from the user irritanterik in #developers on slack.
"insights in last update per sensor / value is a must. I use it to check dead devices. Some devices report only 100% battery and still stop working the next day.
You don't want to lose this info on reboot, so please make this state-full"

**Comment by nattlip** [comment]
i stumbled in this. if you put data in device.data like the id homey remembers it even after app kill.  so you could make there another property lastupdate and last retrieved value which survives app kill.

**Comment by nattlip** [comment]
it even survives a PTP , i checked. but if it survives a firmware update , i dont know.

**Comment by irritanterik** [comment]
Device data data cannot be updated and must be static:
```
All properties in this object are free for you to choose. They should however only be static, not dynamic. So a device's MAC address or UUID makes sense, an IP address does not, because that can change over time. Dynamic properties that are user-changeable (such as a username and password) should be placed in your device's settings, other properties (such as a dynamic IP address) should be discovered upon initialisation of your driver.
```

**Comment by nattlip** [comment]
@irritanterik , you are right. second try. in insights all data are stored with timestamp , would be nice to have acces to them , they are there why not make them accesible and usable by apps.

**Comment by JPe2** [comment]
something like this:
![image](https://cloud.githubusercontent.com/assets/3833862/19222099/f58d6c8a-8e50-11e6-8314-296ffbdfafec.png)

**Comment by WeeJeWel** [comment]
I've added it to the API `GET /manager/devices/device/:id`:

```
"lastUpdated": {
	"onoff": "2016-10-10T13:19:51.986Z"
}
```

I'll think about the UI. On mobile, a tooltip doesn't work, and the interface shouldn't be cluttered as well.

**Comment by irritanterik** [comment]
will this survives reboot?

Op 10 okt. 2016 15:21 schreef "Emile Nijssen" <notifications@github.com>:

**Comment by WeeJeWel** [comment]
Nope. But on a reboot, the app's driver is 'get'-ed for the latest value.

**Comment by irritanterik** [comment]
latest get !== last update

Can we set last update with the cap value get function from the driver?

Op 10 okt. 2016 15:26 schreef "Emile Nijssen" <notifications@github.com>:

**Comment by WeeJeWel** [comment]
Why wouldn't a get be a last update?

**Comment by ewikner** [comment]
Why are you so cautious with get and set functions?

For Mysensors you can get and send a "HEARTBEAT" command to know which sensors is alive and is dead.
I want to have one lastUpdate timestamp and one lastSeen timestamp.
The lastUpdate when the value has changed and lastSeen if the value has changed or if you get a "HEARTBEAT" command.

If you add the possibility to add variables dynamicly with get and set functions then developers can use them in what ever way  suits their app best.

**Comment by irritanterik** [comment]
Ok, get can be last update. but to illustrate, think we need a 'last change' or 'last seen' value. It's relevant to know when the sensor send a changed value. Using the capability get for this will not do the job. All dates will be the same after a reboot.
Maybe I can best illustrate the use case with a prototype. 

![image](https://cloud.githubusercontent.com/assets/15232724/19245077/90f8d9ce-8f1f-11e6-8227-a0266ba1f13a.png)

**Comment by WeeJeWel** [comment]
I see what you want now. Maybe we'll add it someday, but you can imagine that it has not a high priority, as most devices simply don't need it.

**Comment by markaswift** [comment]
Great, so one of the best apps on the App Store (MySensors) suffers because we don't have a timeline. Perhaps we'll get it "someday"... Really @WeeJeWel?

**Comment by WeeJeWel** [comment]
Priorities, priorities... I can't do it all at the same time!

**Comment by ewikner** [comment]
@WeeJeWel: That's sad to read.
To make the app for MySensors to work fully we need this and the "re-pair" function.
But I will try to find an other way for this to work until this will be added.

**Comment by irritanterik** [comment]
Prepared a condition card for the candy app based on the new value in device api:
![image](https://cloud.githubusercontent.com/assets/15232724/19249268/e3949b36-8f34-11e6-8b4b-c890c221bce0.png)
